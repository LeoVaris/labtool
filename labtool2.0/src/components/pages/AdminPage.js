import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader, Label, Table, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router'
import { HorizontalScrollable } from '../HorizontalScrollable'
import { formatCourseName } from '../../util/format'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { sortUsersBySysopLastname } from '../../util/sort'

const AdminCourseList = ({ courses }) => {
  if (!Array.isArray(courses)) {
    return <div />
  }

  return (
    <HorizontalScrollable>
      <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="1">Course ID</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
            <Table.HeaderCell colSpan="2"> </Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {courses.map(instance => (
            <Table.Row key={instance.id}>
              <Table.Cell>{instance.shorterId} </Table.Cell>
              <Table.Cell>
                <strong>{formatCourseName(instance.name, instance.ohid, instance.start)}</strong>
              </Table.Cell>

              <Table.Cell> {instance.europeanStart} </Table.Cell>
              <Table.Cell textAlign="center">
                <Popup
                  trigger={
                    <Button
                      circular
                      size="tiny"
                      icon={{ name: 'edit', size: 'large', color: 'orange' }}
                      as={Link}
                      to={{
                        pathname: `/labtool/ModifyCourseInstanceStaff/${instance.ohid}`,
                        state: { fromAdmin: true }
                      }}
                    />
                  }
                  position="top right"
                  content="Manage instructors for this course"
                />
              </Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </HorizontalScrollable>
  )
}

AdminCourseList.propTypes = {
  courses: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
}

const AdminUserList = ({ me, users, updateOtherUser }) => {
  const setAdmin = (userId, name, newAdmin) => () => {
    const message = `Are you sure you want to ${newAdmin ? 'grant admin rights to' : 'remove admin rights from'} ${name}?`
    if (window.confirm(message)) {
      const selfInflicted = userId === me.user.id
      if (selfInflicted && !newAdmin && !window.confirm('You probably do NOT want to remove admin rights from yourself. Are you really sure?')) {
        return
      }
      updateOtherUser({ id: userId, sysop: newAdmin })
    }
  }
  const sortedUsers = sortUsersBySysopLastname(users)

  return (
    <Table singleLine color="yellow" fixed>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name</Table.HeaderCell>
          <Table.HeaderCell>Status</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {sortedUsers.map(user => (
          <Table.Row key={user.id}>
            <Table.Cell>
              {user.firsts} {user.lastname}
              <br />
              <a href={`mailto:${user.email}`}>{user.email}</a>
            </Table.Cell>
            <Table.Cell>
              {user.teacher && (
                <Label color="orange" horizontal>
                  Teacher
                </Label>
              )}
              {user.sysop ? (
                <div>
                  <Label color="orange" horizontal>
                    Admin
                  </Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, false)} size="tiny" color="green">
                    Remove admin
                  </Button>
                </div>
              ) : (
                <div>
                  <Label horizontal>Not admin</Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, true)} size="tiny" color="green">
                    Make admin
                  </Button>
                </div>
              )}
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  )
}

AdminUserList.propTypes = {
  me: PropTypes.object.isRequired,
  users: PropTypes.array.isRequired,
  updateOtherUser: PropTypes.func.isRequired
}

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.loading.loading) {
    return <Loader active />
  }

  return (
    <div className="AdminPage">
      <h2>Welcome to the admin interface.</h2>
      <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
        {state.coursesList ? 'Hide courses' : 'Manage courses'}
      </Button>
      <br />
      <br />
      {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
      <br />
      <hr />
      <br />
      <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
        {state.usersList ? 'Hide users' : 'Manage users'}
      </Button>
      <br />
      {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage)
