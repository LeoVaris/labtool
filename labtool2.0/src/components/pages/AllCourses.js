import React, { Component } from 'react'
import { getAll } from '../../services/courseInstance'
import { Link } from 'react-router-dom'

class AllCourses extends Component {
  constructor(props) {
    super(props)
    this.state = {
      courses: [],
    }
  }

  componentDidMount() {
    getAll().then(courses =>
      this.setState({ courses })
    )
  }

  render() {
    return (
      <div>
        <ul>
          {
            this.state.courses.map(c => (
              <li key={c.id}>
                <Link to={`/courses/${c.id}`}>{c.name}</Link>
              </li>
            ))
          }
        </ul>
      </div>
    )
  }
}



export default AllCourses