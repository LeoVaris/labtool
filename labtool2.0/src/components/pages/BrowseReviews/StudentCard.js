import React from 'react'
import PropTypes from 'prop-types'
import { Button, Card, Icon, Label, Header, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { createCourseIdWithYearAndTerm } from '../../../util/format'

import RepoLink from '../../RepoLink'

export const StudentCard = ({ student, previousParticipations, handleMarkAsDropped }) => (
  <Card key={student.id} fluid color="yellow" className="studentCard">
    <Card.Content>
      <Header as="h2">
        {student.User.firsts} {student.User.lastname} ({student.User.studentNumber})
        {student.dropped && (
          <Label style={{ marginTop: 5, float: 'right' }}>
            <Icon name="warning sign" color="red" />
            Has dropped course
          </Label>
        )}
        <Header.Subheader>
          <a href={`mailto:${student.User.email}`}>{student.User.email}</a>
        </Header.Subheader>
      </Header>
      <div style={{ display: 'inline-block' }}>
        {student.projectName}: <RepoLink url={student.github} />
        <br />
        {previousParticipations.length > 0 ? (
          <div className="hasOther">
            <p style={{ color: 'red' }}>Has taken this course in other periods</p>
            {previousParticipations.map(participation => (
              <Link key={participation.id} to={`/labtool/browsereviews/${participation.ohid}/${participation.courseInstances[0].id}`}>
                <Popup content="click to see the details" trigger={<p>{createCourseIdWithYearAndTerm(participation.ohid, participation.start)}</p>} />
              </Link>
            ))}
          </div>
        ) : (
          <p className="noOther">Has no other participation in this course</p>
        )}
      </div>
      {
        <Button color="red" style={{ float: 'right' }} onClick={() => handleMarkAsDropped(!student.dropped)}>
          {student.dropped ? 'Mark as non-dropped' : 'Mark as dropped'}
        </Button>
      }
    </Card.Content>
  </Card>
)

StudentCard.propTypes = {
  student: PropTypes.object.isRequired,
  previousParticipations: PropTypes.array.isRequired,
  handleMarkAsDropped: PropTypes.func.isRequired
}

export default StudentCard
