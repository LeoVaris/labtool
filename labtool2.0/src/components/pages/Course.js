import React, { Component } from 'react'
import { getOne } from '../../services/courseInstance'



class Course extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      course: undefined,
    }
  }

  componentDidMount() {
    getOne(parseInt(this.props.match.params.number, 10)).then(
      course => this.setState({ course })
    )
  }

  render() {
    if (!this.state.course) {
      return <div>course not found!!!</div>
    }

    return (
      <div>
        <h1>{this.state.course.name} </h1>
      </div>
    )
  }
}

export default Course