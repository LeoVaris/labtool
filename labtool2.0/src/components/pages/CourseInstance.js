import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li style={{ textAlignVertical: 'center', textAlign: 'center' }}> {instance.name} <button name={instance.name} onClick={handleFieldChange} value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance