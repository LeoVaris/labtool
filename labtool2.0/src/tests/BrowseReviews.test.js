import React from 'react'
import { BrowseReviews } from '../components/pages/BrowseReviews'
import { shallow } from 'enzyme'

describe('<BrowseReviews />', () => {
  let wrapper

  const coursePage = {
    id: 10011,
    name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
    start: '2018-03-11T21:00:00.000Z',
    end: '2018-04-29T21:00:00.000Z',
    active: true,
    weekAmount: 7,
    weekMaxPoints: 3,
    currentWeek: 1,
    ohid: 'TKT20010.2018.K.A.1',
    teacherInstances: [
      {
        id: 10001,
        admin: true,
        userId: 10010,
        courseInstanceId: 10011,
        firsts: 'Pää',
        lastname: 'Opettaja'
      },
      {
        id: 10011,
        admin: true,
        userId: 10015,
        courseInstanceId: 10011,
        firsts: 'Ossi Ohjaaja',
        lastname: 'Mutikainen'
      }
    ]
  }

  const courseData = {
    role: 'teacher',
    data: [
      {
        id: 10012,
        github: 'http://github.com/tiralabra2',
        projectName: 'Tiran toinen labraprojekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10012,
        teacherInstanceId: 10011,
        weeks: [],
        User: {
          id: 10012,
          username: 'tiraopiskelija2',
          email: 'johan.studerande@helsinki.fi',
          firsts: 'Johan Wilhelm',
          lastname: 'Studerande',
          studentNumber: '014553242',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      },
      {
        id: 10031,
        github: 'http://github.com/superprojekti',
        projectName: 'Tira super projekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-06-05T07:12:28.603Z',
        courseInstanceId: 10011,
        userId: 10031,
        teacherInstanceId: 10011,
        weeks: [],
        User: {
          id: 10031,
          username: 'superopiskelija',
          email: 'teras.henkilo@helsinki.fi',
          firsts: 'Teräs',
          lastname: 'Henkilö',
          studentNumber: '014666666',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      },
      {
        id: 10011,
        github: 'http://github.com/tiralabra1',
        projectName: 'Tiran labraprojekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10011,
        teacherInstanceId: 10011,
        weeks: [
          {
            id: 10002,
            points: 2,
            weekNumber: 2,
            feedback: 'Melko hienoa työtä!',
            instructorNotes: 'READMEssa ongelmia',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10001,
            points: 3,
            weekNumber: 1,
            feedback: 'Hienoa työtä!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10003,
            points: 3,
            weekNumber: 3,
            feedback: 'Erittäin hienoa työtä!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10004,
            points: 3,
            weekNumber: 4,
            feedback: 'Hyvin menee!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          }
        ],
        codeReviews: [],
        User: {
          id: 10011,
          username: 'tiraopiskelija1',
          email: 'maarit.opiskelija@helsinki.fi',
          firsts: 'Maarit Mirja',
          lastname: 'Opiskelija',
          studentNumber: '014578343',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      }
    ]
  }

  const loading = {
    loading: false,
    loadingHooks: [],
    redirect: false,
    redirectHooks: [],
    redirectFailure: false
  }

  let mockFn = jest.fn()

  let mockUpdateStudentProjectInfo

  const studentWithoutPreviousParticipation = [coursePage]

  const studentInstanceId = '10011'

  beforeEach(() => {
    mockUpdateStudentProjectInfo = jest.fn()

    wrapper = shallow(
      <BrowseReviews
        getOneCI={mockFn}
        coursePageInformation={mockFn}
        getCoursesByStudentId={mockFn}
        updateStudentProjectInfo={mockUpdateStudentProjectInfo}
        courseData={courseData}
        selectedInstance={coursePage}
        studentInstanceToBeReviewed={studentWithoutPreviousParticipation}
        courseId={coursePage.ohid}
        studentInstance={studentInstanceId} //studentInstance id which was chosen randomly from courseData
        loading={loading}
        resetLoading={mockFn}
        initialLoading={false}
        user={{}}
        createOneComment={mockFn}
        gradeCodeReview={mockFn}
        sendEmail={mockFn}
      />
    )
  })

  describe('BrowseReviews Component', () => {
    it('is ok', () => {
      true
    })

    it('should render without throwing an error', () => {
      expect(wrapper.find('.BrowseReviews').exists()).toEqual(true)
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('student card', () => {
      it('should render student card', () => {
        expect(wrapper.find('.studentCard').exists()).toEqual(true)
      })
      it('when student participates the course first time', () => {
        expect(wrapper.find('.noPrevious').text()).toEqual('Has not taken part in this course before')
      })
      it('when student participate previous instances of the course', () => {
        const studentWithPreviousParticipation = [
          {
            ...coursePage,
            id: 10000,
            name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
            start: '2017-03-11T21:00:00.000Z',
            end: '2017-04-29T21:00:00.000Z',
            active: false,
            ohid: 'TKT20010.2017.K.A.1',
            courseInstances: [{ id: 1 }]
          },
          {
            ...coursePage,
            courseInstances: [{ id: 10011 }]
          }
        ]
        wrapper = shallow(
          <BrowseReviews
            getOneCI={mockFn}
            coursePageInformation={mockFn}
            getCoursesByStudentId={mockFn}
            courseData={courseData}
            selectedInstance={coursePage}
            studentInstanceToBeReviewed={studentWithPreviousParticipation}
            courseId={coursePage.ohid}
            studentInstance={'10011'}
            loading={loading}
            resetLoading={mockFn}
            initialLoading={false}
            updateStudentProjectInfo={mockFn}
            user={{}}
            createOneComment={mockFn}
            gradeCodeReview={mockFn}
            sendEmail={mockFn}
          />
        )
        expect(wrapper.find('.hasPrevious').text()).toContain('Has taken this course before')
        const popup = wrapper
          .find('.hasPrevious')
          .find('Link')
          .find('Popup')
        expect(popup.props()).toHaveProperty('trigger', <p>TKT20010 16-17 4.period</p>)
      })
      it('student can be marked as dropped and non-dropped', () => {
        wrapper.find({ children: 'Mark as dropped' }).simulate('click')

        expect(mockUpdateStudentProjectInfo).toBeCalledWith(expect.objectContaining({ dropped: true }))
      })
    })
  })
})
