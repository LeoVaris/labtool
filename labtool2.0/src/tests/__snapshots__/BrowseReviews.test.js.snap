// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowseReviews /> BrowseReviews Component should render correctly 1`] = `
<div
  className="BrowseReviews"
  style={
    Object {
      "overflowX": "auto",
    }
  }
>
  <Loader
    active={false}
  />
  <div>
    <Connect(BackButton)
      preset="coursePage"
    />
    <Link
      style={
        Object {
          "textAlign": "center",
        }
      }
      to="/labtool/courses/TKT20010.2018.K.A.1"
    >
      <h2>
         
        Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit
         
      </h2>
    </Link>
    <Card
      className="studentCard"
      color="yellow"
      fluid={true}
      key="0"
    >
      <CardContent>
        <h2>
          Maarit Mirja
           
          Opiskelija
           (
          014578343
          )
        </h2>
        <h3>
          <a
            href="mailto:maarit.opiskelija@helsinki.fi"
          >
            maarit.opiskelija@helsinki.fi
          </a>
        </h3>
        <h3>
          Tiran labraprojekti
           
          <a
            href="http://github.com/tiralabra1"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/tiralabra1
          </a>
        </h3>
        <p>
          Has dropped course: false
        </p>
        <Button
          as="button"
          onClick={[Function]}
        >
          Mark as dropped
        </Button>
        <p
          className="noPrevious"
        >
          Has not taken part in this course before
        </p>
      </CardContent>
    </Card>
    <span
      key="1"
    >
      <Button
        as="button"
        onClick={[Function]}
        size="small"
        type="button"
      >
        Show all reviews
      </Button>
      <br />
    </span>
    <Accordion
      fluid={true}
      key="2"
      styled={true}
    >
      <AccordionTitle
        index={0}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 1
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Hienoa työtä!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <h4>
           Comments 
        </h4>
        <CommentGroup />
        <Form
          as="form"
          id={10001}
          name={10001}
          onSubmit={[Function]}
          reply={true}
        >
          <FormMarkdownTextArea
            defaultValue=""
            name="content"
            placeholder="Your comment..."
          />
          <FormCheckbox
            as={[Function]}
            control={[Function]}
            label="Add comment for instructors only"
            name="hidden"
          />
          <Button
            as="button"
            content="Add Reply"
            icon="edit"
            labelPosition="left"
            primary={true}
          />
        </Form>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/1"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="3"
      styled={true}
    >
      <AccordionTitle
        index={1}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 2
        , points 
        2
      </AccordionTitle>
      <AccordionContent>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              2
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Melko hienoa työtä!
            </ReactMarkdown>
             
            <div>
              <br />
              <h4>
                Review notes for instructors 
              </h4>
              <ReactMarkdown
                astPlugins={Array []}
                escapeHtml={true}
                parserOptions={Object {}}
                plugins={Array []}
                rawSourcePos={false}
                renderers={Object {}}
                skipHtml={false}
                sourcePos={false}
                transformLinkUri={[Function]}
              >
                READMEssa ongelmia
              </ReactMarkdown>
               
            </div>
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <h4>
           Comments 
        </h4>
        <CommentGroup />
        <Form
          as="form"
          id={10002}
          name={10002}
          onSubmit={[Function]}
          reply={true}
        >
          <FormMarkdownTextArea
            defaultValue=""
            name="content"
            placeholder="Your comment..."
          />
          <FormCheckbox
            as={[Function]}
            control={[Function]}
            label="Add comment for instructors only"
            name="hidden"
          />
          <Button
            as="button"
            content="Add Reply"
            icon="edit"
            labelPosition="left"
            primary={true}
          />
        </Form>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/2"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="4"
      styled={true}
    >
      <AccordionTitle
        index={2}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 3
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Erittäin hienoa työtä!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <h4>
           Comments 
        </h4>
        <CommentGroup />
        <Form
          as="form"
          id={10003}
          name={10003}
          onSubmit={[Function]}
          reply={true}
        >
          <FormMarkdownTextArea
            defaultValue=""
            name="content"
            placeholder="Your comment..."
          />
          <FormCheckbox
            as={[Function]}
            control={[Function]}
            label="Add comment for instructors only"
            name="hidden"
          />
          <Button
            as="button"
            content="Add Reply"
            icon="edit"
            labelPosition="left"
            primary={true}
          />
        </Form>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/3"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="5"
      styled={true}
    >
      <AccordionTitle
        index={3}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 4
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Hyvin menee!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <h4>
           Comments 
        </h4>
        <CommentGroup />
        <Form
          as="form"
          id={10004}
          name={10004}
          onSubmit={[Function]}
          reply={true}
        >
          <FormMarkdownTextArea
            defaultValue=""
            name="content"
            placeholder="Your comment..."
          />
          <FormCheckbox
            as={[Function]}
            control={[Function]}
            label="Add comment for instructors only"
            name="hidden"
          />
          <Button
            as="button"
            content="Add Reply"
            icon="edit"
            labelPosition="left"
            primary={true}
          />
        </Form>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/4"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="6"
      styled={true}
    >
      <AccordionTitle
        index={4}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 5
         
      </AccordionTitle>
      <AccordionContent>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/5"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="7"
      styled={true}
    >
      <AccordionTitle
        index={5}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 6
         
      </AccordionTitle>
      <AccordionContent>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/6"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      key="8"
      styled={true}
    >
      <AccordionTitle
        index={6}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 7
         
      </AccordionTitle>
      <AccordionContent>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/7"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
      </AccordionContent>
    </Accordion>
  </div>
</div>
`;
